theano/tensor/opt_uncanonicalize.py

```diff
@@ -128,16 +128,16 @@ def local_reshape_dimshuffle(node):
     If a dimshuffle is inside a reshape and does not change the order
     of dimensions, remove it.
     """
-    if node.op == T.reshape:
+    if isinstance(node.op, T.Reshape):
         input_ = node.inputs[0]
         if input_.owner and isinstance(input_.owner.op, DimShuffle):
-            new_order = input_owner.op.new_order
+            new_order = input_.owner.op.new_order
             offset = 0
             for i, dim in enumerate(new_order):
                 if dim == 'x':
                     offset += 1
                     continue
                 elif i != dim + offset:
                     return False
-            return input_.owner.inputs
+            return [T.reshape(input_.owner.inputs[0], tuple(node.inputs[1].owner.inputs))]
```

Comment:
I am not sure about this line. Is the inputs[1] of the reshape always a MakeVector so in order to go grab the new shape tuple you need to go into the inputs of the MakeVector node?